joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/AstGenRunner.scala

```diff
@@ -156,42 +155,70 @@ class AstGenRunner(config: Config) {
-      case _                                                                  => true
+  private def isTranspiledFile(filePath: String): Boolean = {
+    val file = File(filePath)
+    // We ignore files iff:
+    // - they are *.js files and
+    // - they contain a //sourceMappingURL comment or have an associated source map file and
+    // - a file with the same name is located directly next to them
+    lazy val isJsFile            = file.exists && file.extension.contains(".js")
+    lazy val hasSourceMapComment = IOUtils.readLinesInFile(file.path).exists(_.contains("//sourceMappingURL"))
+    lazy val hasSourceMapFile    = File(s"$filePath.map").exists
+    lazy val hasSourceMap        = hasSourceMapComment || hasSourceMapFile
+    lazy val hasFileWithSameName =
+      file.siblings.exists(_.nameWithoutExtension(includeAll = false) == file.nameWithoutExtension)
+    if (isJsFile && hasSourceMap && hasFileWithSameName) {
+      logger.debug(s"'$filePath' ignored by default (seems to be the result of transpilation)")
+      true
+    } else {
+      false
+    }
+
+  }
+
+  private def filterFiles(files: List[String], out: File): List[String] = {
+    files.filter { file =>
+      file.stripSuffix(".json").replace(out.pathAsString, config.inputPath) match {
+        // We are not interested in JS / TS type definition files at this stage.
+        // TODO: maybe we can enable that later on and use the type definitions there
+        //  for enhancing the CPG with additional type information for functions
+        case filePath if TypeDefinitionFileExtensions.exists(filePath.endsWith) => false
+        case filePath if isIgnoredByUserConfig(filePath)                        => false
+        case filePath if isIgnoredByDefault(filePath)                           => false
+        case filePath if isTranspiledFile(filePath)                             => false
+        case _                                                                  => true
+      }
     }
   }
 
   private def processEjsFiles(in: File, out: File, ejsFiles: List[String]): Try[Seq[String]] = {
     val tmpJsFiles = ejsFiles.map { ejsFilePath =>
       val ejsFile           = File(ejsFilePath)
-      val ls                = SourceFiles.retrieveLineSeparator(ejsFilePath)
-      val sourceFileContent = IOUtils.readLinesInFile(ejsFile.path).mkString("", ls, ls)
-      val preprocessContent = new EjsPreprocessor().preprocess(sourceFileContent)
-      (out / in.relativize(ejsFile).toString).parent.createDirectoryIfNotExists(createParents = true)
-      val newEjsFile = ejsFile.copyTo(out / in.relativize(ejsFile).toString)
-      val jsFile     = newEjsFile.changeExtensionTo(".js").writeText(preprocessContent)
-      newEjsFile.createFile().writeText(sourceFileContent)
-      jsFile
+      val maybeOriginalFile = File(ejsFilePath.stripSuffix(".ejs") + ".js")
+      if (isTranspiledFile(maybeOriginalFile.pathAsString)) {
+        maybeOriginalFile
```

Comment:
If I understand things right, then this reuses a transpiled `.js` file if it exists for a `.ejs` template instead of doing our own transpilation? Makes sense to me, but shouldn't the variable referring to the `.js` file than be named sth like `maybeTranspiledFile`?

Reply:
Right, it should 👍